docs/integrations/argocd-integration.md

```diff
@@ -0,0 +1,425 @@
+ 
+```
+ kubectl create secret generic github-secret \
+   --from-file=ssh-privatekey= < path to your private key > \
+    --from-file=ssh-publickey=< path to your public key >
+```
+
+Config-map: 
+
+``` 
+  kubectl  create configmap ssh-known-hosts --from-file=
+   < path to you known hosts file >
+```   
+ 
+ssh config secret
+
+```
+ kubectl create secret generic ssh-config-secret \
+   --from-file=< path to your ssh config file >
+```   
+ 
+Example ssh config file
+
+``` 
+ Host github.com
+  HostName github.com
+  IdentityFile ~/.ssh/id_ed25519_github
+```
+
+Once the presynchook yaml file is completely configured, add this file to your repository folder for which Argo CD pipeline is configured.
+
+
+### 2. Use Terrascan Admission Webhook from PreSyncHook
+___
+You can use the already deployed terrascan’s k8s admission controller webhook to scan the remote repository from Argo CD PreSync hook.
+To configure, follow below steps
+
+
+#### Step 1: Configure terrascan admission controller webhook deployment yaml file with required keys and volumes and service to expose the controller pod.
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+name: terra-controller-webhook
+labels:
+  app: terrascan-admission-webhook
+spec:
+replicas: 1
+selector:
+  matchLabels:
+    app: terrascan-admission-webhook
+template:
+  metadata:
+    labels:
+      app: terrascan-admission-webhook
+  spec:
+    containers:
+    - name: terrascan-admission-webhook
+      image: <TERRASCAN IMAGE>
+      resources:
+        limits:
+          memory: "256Mi"
+          cpu: "1"
+      ports:
+        - containerPort: 443
+      livenessProbe:
+        initialDelaySeconds: 30
+        periodSeconds: 10
+        timeoutSeconds: 5
+        httpGet:
+          path: /health
+          port: 443
+          scheme: HTTPS
+      env:
+        - name: K8S_WEBHOOK_API_KEY
+          value: yoursecretapikey
+      volumeMounts:
+        - mountPath: /data
+          name: certspath
+        - mountPath: /etc/github-secret
+          name: ssh-keys-github
+          readOnly: true
+        - mountPath: /etc/ssh-config-secret
+          name: ssh-config
+          readOnly: true
+        - mountPath: /etc/ssh-known-hosts-secret
+          name: ssh-known-hosts
+          readOnly: true
+      command: ["/bin/sh", "-c"]
+      args:
+        - >
+          cp /etc/github-secret/ssh-private-key /home/terrascan/.ssh/id_ed25519_github &&
+          cp /etc/ssh-config-secret/ssh-config /home/terrascan/.ssh/config &&
+          cp /etc/ssh-known-hosts-secret/ssh-known-hosts /home/terrascan/.ssh/known_hosts &&
+          chmod -R 400 /home/terrascan/.ssh/* &&
+          terrascan server --cert-path /data/server.crt --key-path /data/server.key -p 443 -l debug -c /data/config.toml
+    volumes:
+      - name: certspath
+        hostPath:
+          path: <YOUR CERTIFICATE FOLDER PATH>
```

Comment:
use a secret here instead?

Reply:
added changes to use secret instead of hostpath